Sources/SourceKitLSP/Swift/OpenInterface.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SourceKitD
+import LanguageServerProtocol
+import LSPLogging
+
+struct InterfaceInfo {
+  var contents: String
+}
+
+private var textualInterfaces: [String:DocumentURI] = [:]
+
+extension SwiftLanguageServer {
+  public func openInterface(_ request: LanguageServerProtocol.Request<LanguageServerProtocol.OpenInterfaceRequest>) {
+    let uri = request.params.textDocument.uri
+    let moduleName = request.params.name
+    self.queue.async {
+      
+      if let uri = textualInterfaces[moduleName] {
+        request.reply(.success(InterfaceDetails(uri: uri)))
+        return
+      }
+      
+      let interfaceFilePath = self.generatedInterfacesPath.appendingPathComponent("\(moduleName).swift")
```

Comment:
@ahoppen Should these generated interface files have a `.swiftinterface` extension instead of `.swift`?

Reply:
Oh, yes I think they should.